### PR TITLE
[5.1] Proposal for seeInInput method

### PR DESCRIPTION
**Pull request by sileence**
Here is a similar method that I've used in Codeception and I wasn't able to find in the integration test component.

This works only for input fields, but I could try to implement this for selects, radios, etc. 

Please let me know if you are interested in adding these kind of methods for this package.

### CODE REVIEW

File: src/Illuminate/Foundation/Testing/CrawlerTrait.php [hunk 1]
```diff
@@ -648,6 +648,23 @@ protected function storeInput($element, $text)
     }
 
     /**
+     * Assert that a given string is seen on an input field.
+     *
+     * @param $selector
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Testing/CrawlerTrait.php]: You must provide type information and correct formatting.

File: src/Illuminate/Foundation/Testing/CrawlerTrait.php [hunk 2]
```diff
@@ -648,6 +648,23 @@ protected function storeInput($element, $text)
...
+     * @param $value
+     * @return $this
+     */
+    public function seeInInput($selector, $value)
+    {
+        $this->assertTrue(
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Testing/CrawlerTrait.php]: You should use assertSame here.

Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Testing/CrawlerTrait.php]: assertTrue with a comparison is generally a bad practice

**Comment by Anahkiasen** [comment]
Wouldn't there be a way to pull in Codeception's module and use that? Cause we're basically just reinventing Codeception piece by piece, probably introducing errors along the way while Codeception's battle tested

**Comment by sileence** [comment]
@Anahkiasen I am not sure about that, I really like Codeception though, but since I was teaching how to use the integration component, I noticed a few missing methods and wanted to contribute them. 

But maybe it's a topic worth of discussion.

In the meantime, I've uploaded some fixes and a test... cc @GrahamCampbell please check and thank you for reviewing this.

**Comment by GrahamCampbell** [comment]
Looks ok to me in theory. // cc @taylorotwell

**Comment by sileence** [comment]
Maybe also @JeffreyWay could give us an opinion here. 

If you decide this is good, I'll work on creating the other methods like seeInField or isSelected, isChecked, etc.

**Comment by taylorotwell** [comment]
Why don't you flesh it out with the other input types and I'll take a look.

**Comment by sileence** [comment]
@taylorotwell ok, I'll work on that.

**Comment by GrahamCampbell** [comment]
Could you rebase this please.

**Comment by GrahamCampbell** [comment]
And squash to one commit.

File: src/Illuminate/Foundation/Testing/CrawlerTrait.php [hunk 3]
```diff
@@ -648,6 +648,53 @@ protected function storeInput($element, $text)
     }
 
     /**
+     * Assert that an input field contains the given value.
+     *
+     * @param  string $selector
+     * @param  mixed  $expected
+     * @return $this
+     * @throws \Exception
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Testing/CrawlerTrait.php]: you need a newline before this

File: src/Illuminate/Foundation/Testing/CrawlerTrait.php [hunk 4]
```diff
@@ -648,6 +648,53 @@ protected function storeInput($element, $text)
...
+
+        return $this;
+    }
+
+    /**
+     * Get an input or textarea value.
+     *
+     * @param  $selector
+     * @return string
+     * @throws \Exception
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Testing/CrawlerTrait.php]: same

File: src/Illuminate/Foundation/Testing/CrawlerTrait.php [hunk 5]
```diff
@@ -648,6 +648,53 @@ protected function storeInput($element, $text)
...
+     */
+    protected function getInputOrTextareaValue($selector)
+    {
+        $field = $this->filterByNameOrId($selector);
+
+        if ($field->count() == 0) {
+            throw new \Exception("There are no elements with the name or ID [$selector]");
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Testing/CrawlerTrait.php]: import this please

File: src/Illuminate/Foundation/Testing/CrawlerTrait.php [hunk 6]
```diff
@@ -648,6 +648,53 @@ protected function storeInput($element, $text)
...
+            "The input [{$selector}] has not the value [{$expected}]."
+        );
+
+        return $this;
+    }
+
+    /**
+     * Get an input or textarea value.
+     *
+     * @param  $selector
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Testing/CrawlerTrait.php]: wrong

File: tests/Foundation/FoundationCrawlerTraitTest.php [hunk 7]
```diff
@@ -12,6 +12,63 @@ public function tearDown()
         m::close();
     }
 
+    public function testSeeInFieldInput()
+    {
+        $input = m::mock('\Symfony\Component\DomCrawler\Crawler')->makePartial();
```
Comment (GrahamCampbell) [inline tests/Foundation/FoundationCrawlerTraitTest.php]: no slash

File: tests/Foundation/FoundationCrawlerTraitTest.php [hunk 8]
```diff
@@ -12,6 +12,63 @@ public function tearDown()
...
+        $input->shouldReceive('count')->andReturn(1);
+        $input->shouldReceive('nodeName')->once()->andReturn('input');
+        $input->shouldReceive('attr')
+            ->withArgs(['value'])
+            ->once()
+            ->andReturn('Laravel');
+
+        $this->crawler = m::mock('\Symfony\Component\DomCrawler\Crawler')->makePartial();
```
Comment (GrahamCampbell) [inline tests/Foundation/FoundationCrawlerTraitTest.php]: same

File: tests/Foundation/FoundationCrawlerTraitTest.php [hunk 9]
```diff
@@ -12,6 +12,63 @@ public function tearDown()
...
+            ->withArgs(["*#framework, *[name='framework']"])
+            ->once()
+            ->andReturn($input);
+
+        $this->seeInField('framework', 'Laravel');
+    }
+
+    public function testSeeInFieldTextarea()
+    {
+        $textarea = m::mock('\Symfony\Component\DomCrawler\Crawler')->makePartial();
```
Comment (GrahamCampbell) [inline tests/Foundation/FoundationCrawlerTraitTest.php]: same

File: src/Illuminate/Foundation/Testing/CrawlerTrait.php (hunk 3 above)
Comment (sileence) [inline src/Illuminate/Foundation/Testing/CrawlerTrait.php]: I'm sorry I don't understand what I need to modify here

Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Testing/CrawlerTrait.php]:
```
 *
 * @throws
```

**Comment by GrahamCampbell** [comment]
You rebased against the wrong branch...

**Comment by sileence** [comment]
=/ You are right, but this doesn't look either, should I just create a new pull request? 

This method now covers inputs and textareas, I still need to work on radios, selects and checkboxes.

**Comment by GrahamCampbell** [comment]
If you want.

**Comment by sileence** [comment]
Yes, I'll close this and create a new PR.

**Pull request closed without merging**